src/LogicApp/LogicApp/Cmdlets/IntegrationAccount/NewAzureIntegrationAccountBatchConfigurationCommand.cs

```diff
@@ -0,0 +1,242 @@
+        [Parameter(Mandatory = true, HelpMessage = Constants.BatchConfigurationResourceIdHelpMessage, ParameterSetName = ParameterSet.ByResourceIdAndParameters, ValueFromPipelineByPropertyName = true)]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        #endregion Input Parameters
+
+        /// <summary>
+        /// Executes the integration account batch configuration create command.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            switch (this.ParameterSetName)
+            {
+                case ParameterSet.ByInputObjectAndJson:
+                case ParameterSet.ByInputObjectAndFilePath:
+                case ParameterSet.ByInputObjectAndParameters:
+                {
+                    var parsedResourceId = new ResourceIdentifier(this.ParentObject.Id);
+                    this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                    this.ParentName = parsedResourceId.ResourceName;
+                    break;
+                }
+                case ParameterSet.ByResourceIdAndJson:
+                case ParameterSet.ByResourceIdAndFilePath:
+                case ParameterSet.ByResourceIdAndParameters:
+                {
+                    var parsedResourceId = new ResourceIdentifier(this.ParentResourceId);
+                    this.ResourceGroupName = parsedResourceId.ResourceGroupName;
+                    this.ParentName = parsedResourceId.ResourceName;
+                    break;
+                }
+            }
+
+            var batchConfiguration = new BatchConfiguration();
+
+            switch (this.ParameterSetName)
+            {
+                case ParameterSet.ByInputObjectAndJson:
+                case ParameterSet.ByResourceIdAndJson:
+                case ParameterSet.ByIntegrationAccountAndJson:
+                {
+                    batchConfiguration.Properties = CmdletHelper.ConvertToBatchConfigurationProperties(this.BatchConfigurationDefinition);
+                    break;
+                }
+                case ParameterSet.ByInputObjectAndFilePath:
+                case ParameterSet.ByResourceIdAndFilePath:
+                case ParameterSet.ByIntegrationAccountAndFilePath:
+                {
+                    batchConfiguration.Properties = CmdletHelper.ConvertToBatchConfigurationProperties(CmdletHelper.GetStringContentFromFile(this.TryResolvePath(this.BatchConfigurationFilePath)));
+                    break;
+                }
+                case ParameterSet.ByInputObjectAndParameters:
+                case ParameterSet.ByResourceIdAndParameters:
+                case ParameterSet.ByIntegrationAccountAndParameters:
+                {
+                    var releaseCriteria = new BatchReleaseCriteria();
+                    if (this.MessageCount > 0)
```

Comment:
Instead of doing this validation here, please add a [ValidateRange(0, int.MaxValue)] for MessageCount, BatchSize, and ScheduleInterval

Reply:
This isn't validation. When the parameter isn't specified the default value is set, I'm checking to see if the optional parameter was specified.

Reply:
Shouldn't you be checking if the parameter is set rather than is greater than 0? Currently there is nothing preventing the user from providing a value less than zero.  You can check if that parameter is set using ```this.MyInvocation.BoundParameters.ContainsKey("MessageCount")```

Reply:
I'll move over and use that and add the validation in the inputs section